src/js/App/GlobalFilter/TagsModal.js

```diff
@@ -1,20 +1,37 @@
 import React, { useState, useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import { useSelector, useDispatch } from 'react-redux';
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
 import { TagModal } from '@redhat-cloud-services/frontend-components/components/cjs/TagModal';
 import { fetchAllTags } from '../../redux/actions';
 import debounce from 'lodash/debounce';
 import flatMap from 'lodash/flatMap';
+import { getAllSIDs } from './tagsApi';
 
 const TagsModal = ({ isOpen, filterTagsBy, onApplyTags, toggleModal, selectedTags }) => {
-  const [selected, setSelected] = useState([]);
+  const [tagsSelected, setTagsSelected] = useState([]);
+  const [sidsSelected, setSidsSelected] = useState([]);
   const [filterBy, setFilterBy] = useState('');
   const dispatch = useDispatch();
-  const loaded = useSelector(({ globalFilter: { tags } }) => tags?.isLoaded);
+  const { tagsLoaded, tagsCount, tagsPage, tagsPerPage } = useSelector(
+    ({ globalFilter: { tags } }) => ({
+      tagsLoaded: tags?.isLoaded,
+      tagsCount: tags?.total || 0,
+      tagsPage: tags?.page || 1,
+      tagsPerPage: tags?.perPage || 10,
+    }),
+    shallowEqual
+  );
+  const { sidLoaded, sidCount, sidPage, sidPerPage } = useSelector(
+    ({ globalFilter: { sid } }) => ({
+      sidLoaded: sid?.isLoaded,
+      sidCount: sid?.total || 0,
+      sidPage: sid?.page || 1,
+      sidPerPage: sid?.perPage || 10,
+    }),
+    shallowEqual
+  );
```

Comment:
These two selectors are exactly the same. On the prefix is different. We could use a common function for it.

Reply:
Good point.